### DISCUSSION TITLE
How to read command output when in Control mode (-CC)

### DISCUSSION THREAD

**Discussion by Tomiyou** [tmux/tmux#4016] [General]
I am spawning a new Tmux subprocess in Control mode from a Rust program like this:
```
tmux -CC attach-session -t test_session
```
I need to use two -CC flags, since if I only use a single -C flag, tmux will immediately detach when STDIN is piped to the original Rust program.

However, I don't know how to get output of commands such as `list-windows` as tmux will not print anything to the STDOUT when using -CC. Is there a socket I am supposed to attach to and read output from there? Or am I misunderstanding this whole approach?

**Comment by nicm** [comment]
`-CC` probably only makes sense if you're going to run it in a pty, if you just want to use stdin and stdout as pipes then you're better off with `-C`, although you will need both stdin and stdout.

**Reply by Tomiyou** [reply]
Makes sense, I will use `-C` instead.
